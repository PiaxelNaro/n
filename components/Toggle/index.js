// @flow
import styled, { css } from 'styled-components';

const Toggle = styled.span`
    display: inline-block;
    position: relative;
    width: 26px;
    height: 12px;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${props => (props.checked ? '#3f88d4' : '#efefef')};
    border-radius: 6px;
    ${props => (props.disabled ? css`pointer-events: none;` : '')}
    ::before {
        position: absolute;
        top: -1px;
        left: 0;
        width: 14px;
        height: 14px;
        border-radius: 50%;
        background-color: ${props => (props.disabled ? '#ccc' : '#fff')};
        content: "";
        box-shadow: rgba(0,0,0,0.3) 0 1px 3px 0.2px;
        transition: .4s;
        ${props => (props.checked ? css`transform: translateX(12px);` : '')}
    }
`;
Toggle.displayName = 'Toggle';

export default Toggle;
