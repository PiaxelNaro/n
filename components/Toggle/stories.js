import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Toggle from '.';

class Demo extends React.Component {
    constructor() {
        super();

        this.state = {
            checked: false,
        };
        this.toggleHandler = this.toggleHandler.bind(this);
    }

    toggleHandler() {
        this.setState(prevState => ({
            checked: !prevState.checked,
        }));
    }

    render() {
        return (
            <div>
                <p>On (static): <Toggle checked /></p>
                <p>Off (static): <Toggle checked={false} /></p>
                <p>Disabled: <Toggle disabled /></p>
                <p>
                    Toggle:
                    <Toggle onClick={this.toggleHandler} checked={this.state.checked} />
                </p>
            </div>
        );
    }
}

storiesOf('Toggle', module).add('render toggle', () => <Demo />);
