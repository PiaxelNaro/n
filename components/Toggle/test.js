// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Toggle from '.';

describe('<Toggle>', () => {
    it('renders toggle', () => {
        const spy = jest.fn();
        const toggle = shallow(<Toggle onClick={spy} />);
        expect(toggle).toMatchSnapshot();

        toggle.simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);

        toggle.setProps({
            checked: true,
        });
        toggle.update();
        expect(toggle).toMatchSnapshot();
        expect(toggle.prop('checked')).toBeTruthy();

        toggle.setProps({
            disabled: true,
        });
        toggle.update();
        expect(toggle).toMatchSnapshot();
        expect(toggle.prop('disabled')).toBeTruthy();
    });
});
