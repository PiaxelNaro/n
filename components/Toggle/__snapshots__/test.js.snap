// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle> renders toggle 1`] = `
<span
  className="Toggle__Toggle-h8sldt-0 jKEBzD"
  onClick={[Function]}
/>
`;

exports[`<Toggle> renders toggle 2`] = `
<span
  checked={true}
  className="Toggle__Toggle-h8sldt-0 gZIbjx"
  onClick={[Function]}
/>
`;

exports[`<Toggle> renders toggle 3`] = `
<span
  checked={true}
  className="Toggle__Toggle-h8sldt-0 bYSeek"
  disabled={true}
  onClick={[Function]}
/>
`;
